File: lint-arwaky/src-python/contract/metrics_provider_port.py
"""metrics_provider_port — Interface for technical metrics and history."""

from abc import ABC, abstractmethod

from ..taxonomy import FilePath, Count, ResponseDataList, MetricsError


class IMetricsProviderPort(ABC):
    """Port for retrieving code metrics and historical data."""

    @abstractmethod
    async def get_line_count(self, path: FilePath) -> Count | MetricsError:
        """Return the raw line count of a file."""
        ...

    @abstractmethod
    async def get_history(self) -> ResponseDataList | MetricsError:
        """Read the raw quality history log."""
        ...

File: lint-arwaky/src-rust/contract/metrics_provider_port.rs
use async_trait::async_trait;
use super::*;

#[async_trait]
pub trait IMetricsProviderPort: Send + Sync {
    async fn get_line_count(&self, path: &FilePath) -> Count;
    async fn get_history(&self) -> Vec<serde_json::Value>;
}
